driver/src/main/java/org/kaazing/k3po/driver/internal/Robot.java

```diff
@@ -284,23 +287,41 @@ public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws
         return new CompositeChannelFuture<>(channel, bindFutures);
     }
 
-    private void startConfiguration() {
+    private void startConfiguration() throws Exception {
         /* Connect to any clients */
-        for (ClientBootstrap client : configuration.getClientBootstraps()) {
-
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("[id:           ] connect " + client.getOption("remoteAddress"));
+        for (final ClientBootstrapResolver clientBootstrapResolver : configuration.getClientBootstrapResolvers()) {
+            Barrier barrier = clientBootstrapResolver.getBarrier();
+            if (barrier != null) {
+                barrier.getFuture().addListener(new ChannelFutureListener() {
+
+                    @Override
+                    public void operationComplete(ChannelFuture future) throws Exception {
+                        // TODO: check if script is already aborted or script execution has failed
```

Comment:
Is this TODO necessary,  Reading the stopConfiguration() method on Robot.java makes me think that the worst that happens is its fired and then removed, but please confirm